src/vs/base/common/glob.ts

```diff
@@ -817,3 +817,108 @@ export function patternsEquals(patternsA: Array<string | IRelativePattern> | und
 		return false;
 	});
 }
+
+// brace expansion
+
+/**
+ * Split string given first opportunity for brace expansion in the string.
+ * - If the brace is prepended by a \ character, then it is escaped.
+ * - Does not process escapes that are within the sub-glob.
+ * - If two unescaped `{` occur before `}`, then ripgrep will return an error for brace nesting, so don't split on those.
+ */
+function getEscapeAwareSplitString(pattern: string): { fixedStart?: string; strInBraces: string; fixedEnd?: string } {
+	let inBraces = false;
+	let escaped = false;
+	let fixedStart = '';
+	let strInBraces = '';
+	for (let i = 0; i < pattern.length; i++) {
+		const char = pattern[i];
+		switch (char) {
+			case '\\':
+				if (escaped) {
+					// If we're already escaped, then just leave the escaped slash and the preceeding slash that escapes it.
+					// The two escaped slashes will result in a single slash and whatever processes the glob later will properly process the escape
+					if (inBraces) {
+						strInBraces += '\\' + char;
+					} else {
+						fixedStart += '\\' + char;
+					}
+					escaped = false;
+				} else {
+					escaped = true;
+				}
+				break;
+			case '{':
+				if (escaped) {
+					// if we escaped this opening bracket, then it is to be taken literally. Remove the `\` because we've acknowleged it and add the `{` to the appropriate string
+					if (inBraces) {
+						strInBraces += char;
+					} else {
+						fixedStart += char;
+					}
+					escaped = false;
+				} else {
+					if (inBraces) {
+						// ripgrep treats this as attempting to do an alternating group, which is invalid. Return with pattern including changes from escaped braces.
```

Comment:
Just to be clear, this shouldn't be a ripgrep-specific function. glob.ts is our implementation of glob patterns and we should do what's right for our implementation. If it's really "expandBracePatternsForRipgrep" then it can live somewhere else